extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/cors/CORSFilter.java

```diff
@@ -221,6 +221,11 @@ public void handle(RoutingContext event) {
                 response.setStatusCode(403);
                 response.setStatusMessage("CORS Rejected - Invalid origin");
                 response.end();
+            } else if (!allowsMethod) {
+                LOG.debug("Method is not allowed");
```

Comment:
This is confusing a preflight request with the actual request itself. For the actual request you check the method, there will be no 'ACCESS_CONTROL_REQUEST_METHOD' header.

In general this existing code is not ideal. It is applying preflight checks for access control headers to ever request, which leads to this kind of confusion. 

This code needs to be refactored to:

- Split out the normal request and preflight code paths
- in the normal request code path check the actual method (and the headers), and return a 403 if there is a failure
- In the preflight path just return all configured methods and headers, there is not any need to actually parse the Access-Control-Request-Headers or Access-Control-Request-Methods header, all we need to check is if they are present or not.

Reply:
Sure, thanks for the refactoring